File: src/main/frontend/app-prograo/src/app/interfaces/FreelancerDetails.ts
import { CalificationDTO } from "./CalificationDTO";
import { FreelancerDetailsSkills } from "./FreelancerDetailsSkills";
import { FreelancerSkill } from "./FreelancerSkill";

export interface FreelancerDetails {
    userId: number;
    userName: string;
    userProfilePhoto: string;
    userTitle: string;
    userCity: string;
    userCountry: string;
    freelancerId: number;
    freelancerRate: number;
    freelancerDescription: string;
    freelancerTwitter: string;
    freelancerFacebook: string;
    freelancerEmail: string;
    freelancerLinkedin: string;
    calificationList: CalificationDTO[];
    skillList: FreelancerDetailsSkills[];
    totalCalification: number;
    numberCalification: number;
    [key: string]: any;
}

File: src/main/java/com/prograo/dto/FreelancerDetails.java
package com.prograo.dto;

import com.prograo.domain.Freelancer;
import com.prograo.domain.Location;
import com.prograo.domain.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class FreelancerDetails {

    private Long userId;

    private String userName;

    private String userProfilePhoto;

    private String userTitle;

    private String userCity;

    private String userCountry;

    private Long freelancerId;

    private int freelancerRate;

    private String freelancerDescription;

    private String freelancerTwitter;

    private String freelancerFacebook;

    private String freelancerEmail;

    private String freelancerLinkedin;

    private List<CalificationDTO> calificationList;

    private List<SkillUsed> skillList;

    private Double totalCalification;

    private Integer numberCalification;

    public void setFreelancerData(Freelancer freelancer) {
        this.freelancerId = freelancer.getId();
        this.freelancerRate = freelancer.getRate();
        this.freelancerDescription = freelancer.getDescription();
        this.freelancerTwitter = freelancer.getTwitter();
        this.freelancerFacebook = freelancer.getFacebook();
        this.freelancerEmail = freelancer.getEmail();
        this.freelancerLinkedin = freelancer.getLinkedin();
    }

    public void setUserData(User user) {
        this.userId = user.getId();
        this.userName = user.getName();
        this.userProfilePhoto = user.getProfilePhoto();
        this.userTitle = user.getTitle();
    }

    public void setLocationData(Location location) {
        this.userCity = location.getCity();
        this.userCountry = location.getCountry();
    }
}
